### DISCUSSION TITLE
[Feature] Automatic stack suggestions

### DISCUSSION THREAD

**Discussion by Hely0n** [immich-app/immich#6521] [Feature Request]
### The feature

After I imported my whole library, including DSLR photos made by my dad on events, I realized, how much space in my timeline is wasted by "the same photo but shot 8 times". 
The stack feature is phenomenal, but it's a unreasonable undertaking to group every potential stack manually if you have many photos. Not to mention picking the perfect one of those and deleting the others.

So I got that idea, that an algorithm will collect all groups of photos, where the difference of the timestamps don't exceed say 15 sec. Than it will try to cluster the images by some image-difference algorithm (like skimage.structural_similarity) or maybe even a neural network for this purpose. The user can tune the maximum difference to consider.

Because no difference algorithm is perfect, I think it's not a good idea to just group them to stacks automatically, but rather suggest them somehow to the user, so that he/her can stack them with one click.

I still don't have a concrete idea how and where to put it in the UI, but maybe a separate page can be created, where just the suggested stacks and some of their neighbour images are shown. 

This feature should only exist on the web ui, as it's not something you need on a regular basis.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by flightmansam** [comment]
I think this is an absolutely stunning idea and would make so much use of it.

**Comment by bo0tzz** [comment]
Duplicate of #6287 and/or #1968

**Comment by MWP** [comment]
Yes please!
Auto-stack by time would be brilliant.
For example: auto stack photos (on import, or by manual job) that are taken within 30sec of each other.
